### ISSUE TITLE
Excess Inventory slowing down auto-loot

### ISSUE THREAD

**Issue by SteveTownsend** [SteveTownsend/SmartHarvestSE#359]
[Nexus report](https://forums.nexusmods.com/index.php?showtopic=8799893/#entry103806693)

> I activated loot transfer destinations on several different containers appropriate for the various items I wanted sorted. Everything chugged down with a metaphorical whine. I reloaded to an earlier save and did some poking, in which I found that the "handle excess crafting items" for me seems to be slowing down the picking up of items quite a bit. I have not tried to use the destinations since, I admit, so I cannot know atm if that was another reason. Now, a short time later, I've gotten "used to" things being slightly slower.

**Comment by SteveTownsend** [comment]
Script logic to reset _EquippedOrWorn_ is costly for every UI exit - logging DLL shows
```
2022-01-08 14:07:27.907     info  32552 Reset EquippedOrWorn
2022-01-08 14:07:27.950     info  34244 Miner's Clothes/0x00080697 added to EquippedOrWorn
2022-01-08 14:07:27.971     info  32552 Brown Gloves/0x000261c1 added to EquippedOrWorn
2022-01-08 14:07:28.014     info  19476 Miner's Boots/0x00080699 added to EquippedOrWorn
```

**Comment by SteveTownsend** [comment]
The check for excess-handling config should be done upfront in _ContainerLister::CacheIfExcessHandlingEnabled_ - this avoids a ton of per-item secondary checks that are only needed if item is eligible for excess-handling, helping scaling with large inventories.

**Comment by SteveTownsend** [comment]
This seems really expensive in `PlayerState::ItemHeadroom` - why do we recheck entire inventory because we updated/added certain items?
```
	// Trigger reconciliation before next loot scan whenever we loot an item with inventory limits
	m_refreshCache = true;
```

**Comment by SteveTownsend** [comment]
The inefficiency in `PlayerState::ItemHeadroom` would cause what reporting user sees,
1. Enabling crafting-excess likely means more items are now eligible
2. Smart Harvesters tend to be pack rats, with large inventories
3. Each time an eligible item is auto-looted this triggers a full inventory review for excess handling
This combination of factors means a vicious circle: more excess items -> more inventory review s -> slower auto-looting. I think 3. can be addressed by:
- tracking the added/updated eligible items
- processing only those items on interim inventory review
- only triggering a full review periodically or deterministically when we know we should e.g. when MCM closes, meaning possible config changes.
This may result in some edge case artifacts but better than slowness in the service of perfect accuracy.

**Comment by SteveTownsend** [comment]
All this is fixed for v4.1.0.0 - more work may be needed based on user experience in the wild
